### DISCUSSION TITLE
Liquid objects available on NotFound.liquid

### DISCUSSION THREAD

**Discussion by bashuss** [OrchardCMS/OrchardCore#11616] [Q&A]
I am trying to build an intelligent 404 page, as the site I build has been migrated from a different CMS with different paths.
So I would like to analyse the Request.path that leads to the 404 and provide some search results according to this.

I may be able to know wether it is a request to a page from the old CMS or just something stupid by analyzing the pattern of the url.

Unfortunately the liquid {{ Request.path }} does not return the url of the 404 causing request, but "/Error/404", which is not helpful.

1. What information is available from liquid on this page and how?
2. If not, Is there an easy way to provide the information for liquid in a custom model?
3. If I need to do this server-side with JavaScript, how do I get the search results from there?
4. Any thoughts on why it could be a stupid idea trying to create such an intelligent 404 page? Security considerations? 

As a user, I find it highly annoying to get 404 pages, that do not tell me more than just - ERROR. But I know, that there is more to it than just the users point of view.

Any thoughts or knowledge appreciated.
best regards,
Bernd

**Comment by ns8482e** [comment]
Diagnostic feature use StatusCodePagesMiddleware so 

You can get original path via IStatusCodeReExecuteFeature feature in C#/cshtml as following
```cs
var feature = HttpContext.Features.Get<IStatusCodeReExecuteFeature>();
        var originalPath = feature?.OriginalPath;
```

Not sure if there is any accessor available for this in liquid
